Getting Started with Form (#1485)
DevExpress/devextreme-documentation: concepts/05 Widgets/Form/01 Getting Started with Form/30 Configure Simple Items.md

> The [items[]](/Documentation/ApiReference/UI_Widgets/dxForm/Configuration/#items) array is used to configure form items. This array can contain strings and objects. Strings are names of the [formData](/Documentation/ApiReference/UI_Widgets/dxForm/Configuration/#formData) object's fields. Use strings for the fields that the Form will render as simple items with default configuration. You can also configure editors for certain fields manually. For this, specify editor settings in an object in the **items[]** array. Use the [dataField](/Documentation/ApiReference/UI_Widgets/dxForm/Item_Types/SimpleItem/#dataField) property to identify the field that provides data. In the example below, we configure the `HireDate` field's editor:

Reviewer: Your version does deliver the information about strings but makes it look like the objects are for configuring editors only, which isn't true. Also, the difference in use-cases between strings and objects isn't as clear as it could be. Here's how I suggest we fix that:

```suggestion
Use the [items[]](/Documentation/ApiReference/UI_Widgets/dxForm/Configuration/#items) array to configure all form items. This array can contain strings (**formData** field names) and objects (item configurations). Use a string if the default item configuration suits you. Otherwise, declare an object that configures the simple item. Specify the [dataField](/Documentation/ApiReference/UI_Widgets/dxForm/Item_Types/SimpleItem/#dataField) and other properties in it. In the example below, we configure the `HireDate` simple item:
```

            officeNumber: 901,
            hireDate: new Date(2012, 4, 13)
        }

        hireDateOptions = {
            disabled: true
        }
    }

    <!-- tab: app.module.ts -->
    import { BrowserModule } from '@angular/platform-browser';
    import { NgModule } from '@angular/core';
    import { AppComponent } from './app.component';

    import { DxFormModule } from 'devextreme-angular';

    @NgModule({
        declarations: [
            AppComponent
        ],
        imports: [
            BrowserModule,
            DxFormModule
        ],
        providers: [ ],
        bootstrap: [AppComponent]
    })
    export class AppModule { }

##### Vue

    <!-- tab: App.vue -->
    <template>
        <DxForm 
            :form-data="employee">
            <DxItem data-field="name"/>
            <DxItem data-field="officeNumber"/>
            <DxItem 
                data-field="hireDate"
                :editor-options="hireDateOptions"
            />
        </DxForm>
    </template>

    <script>
    import 'devextreme/dist/css/dx.common.css';
    import 'devextreme/dist/css/dx.light.css';

    import { DxForm, DxItem } from 'devextreme-vue/form';
    

>     let employee = {

Reviewer 2: ```suggestion
    const employee = {
```

        name: "John Heart",
        officeNumber: 901,
        hireDate: new Date(2012, 4, 13)
    };

    const hireDateOptions = {
        disabled: true
    };

    export default {
        components: {
            DxForm,
            DxItem
        },
        data: {
            return: {
                employee,
                hireDateOptions
            }
        }
    }
    </script>

##### React

    <!-- tab: App.js -->
    import React from 'react';
    import 'devextreme/dist/css/dx.common.css';
    import 'devextreme/dist/css/dx.light.css';

    import {
        Form,
        Item
    } from 'devextreme-react/form';

>     let employee = {

Reviewer 2: ```suggestion
    const employee = {
```

> To generate items only for the required fields, declare the [items]() array. Each item in this array is a string or an object. A string is a name of a field in the [formData]() object. The **Form** will generate a simple item based on this field's value. To customize an item, specify required properties in an object. Use the [dataField]() property to identify the field that provides data for the item. In this example, we [disable]() the editor for the `hireDate` item:

Reviewer: > To generate items only for the required fields

This is one of possible use-cases, but not the main one. The items array is used to configure form items.

This paragraph also contains many noun "specifiers" and adverbial phrases ("items **only for the required fields**", "Each item **in this array**", "a name **of a field in the [formData]() object**", "specify required properties **in an object**", "the field **that provides data for the item**", "the editor **for the `hireDate` item**"). They make the text hard to follow.

Try to follow this structure:

What the items array is used for.
Which members it can contain.
When to use strings, when to use objects.
How to use objects.

---
##### jQuery

    <!-- tab: index.js -->
    $(function() {
        $("#form").dxForm({
            formData: {
                name: "John Heart",
                officeNumber: "901",
                hireDate: new Date(2012, 4, 13)
            },
            items: ["name", "officeNumber", {
                dataField: "hireDate",
                editorOptions: {
                    disabled: true
                }
            }]
        });
    });

##### Angular

    <!-- tab: app.component.html -->
    <dx-form
        [(formData)]="employee"

>     >

Reviewer: ```suggestion
        [(formData)]="employee">
```

        <dxi-item dataField="name"></dxi-item>
        <dxi-item dataField="officeNumber"></dxi-item>
        <dxi-item 
            dataField="hireDate" 
            [editorOptions]="dateItemOptions"

>         >

Reviewer: ```suggestion
            [editorOptions]="hireDateOptions">
```

        </dxi-item>
    </dx-form>

    <!-- tab: app.component.ts -->
    import { Component } from '@angular/core';

    @Component({
        selector: 'app-root',
        templateUrl: './app.component.html',
        styleUrls: ['./app.component.css']
    })
    export class AppComponent {
        employee = {
            name: "John Heart",
            officeNumber: "901",
            hireDate: new Date(2012, 4, 13)
        }

>         dateItemOptions = {

Reviewer: ```suggestion
        hireDateOptions = {
```

            disabled: true

>         };

Reviewer: ```suggestion
        }
```

    }

    <!-- tab: app.module.ts -->
    import { BrowserModule } from '@angular/platform-browser';
    import { NgModule } from '@angular/core';
    import { AppComponent } from './app.component';

    import { DxFormModule } from 'devextreme-angular';

    @NgModule({
        declarations: [
            AppComponent
        ],
        imports: [
            BrowserModule,
            DxFormModule
        ],
        providers: [ ],
        bootstrap: [AppComponent]
    })
    export class AppModule { }

##### Vue

    <!-- tab: App.vue -->
    <template>
        <DxForm 
            :form-data="employee">
            <DxItem data-field="name"/>
            <DxItem data-field="officeNumber"/>
            <DxItem 
                data-field="hireDate"

>                 :editor-options="dateItemOptions"

Reviewer: ```suggestion
                :editor-options="hireDateOptions"
```

            />
        </DxForm>
    </template>

    <script>
    import 'devextreme/dist/css/dx.common.css';
    import 'devextreme/dist/css/dx.light.css';

    import { DxForm, DxItem } from 'devextreme-vue/form';
    

>     const employee = {

Reviewer: This object was declared inside the component in the previous step.

        name: "John Heart",
        officeNumber: "901",
        hireDate: new Date(2012, 4, 13)
    };

>     const dateItemOptions = {

Reviewer: ```suggestion
    const hireDateOptions = {
```

        disabled: true
    };

    export default {
        components: {
            DxForm,
            DxItem
        },
        data: {
            return: {
                employee,

>                 dateItemOptions

Reviewer: ```suggestion
                hireDateOptions
```

            }
        }
    }
    </script>

##### React

    <!-- tab: App.js -->
    import React from 'react';
    import 'devextreme/dist/css/dx.common.css';
    import 'devextreme/dist/css/dx.light.css';

    import {
        Form,
        Item
    } from 'devextreme-react/form';

>     const employee = {

Reviewer: Same note about the constant.

        name: "John Heart",
        officeNumber: "901",
        hireDate: new Date(2012, 4, 13)
    };

>     const dateItemOptions = {

Reviewer: ```suggestion
    const hireDateOptions = {
```

        disabled: true
    };

    const App = () => {
        return (
            <Form
                formData={employee}>
                <Item dataField="name" />
                <Item dataField="officeNumber" />
                <Item 
                    dataField="hireDate"

>                     editorOptions={dateItemOptions}

Reviewer: ```suggestion
                    editorOptions={hireDateOptions}
```
